providers/implementations/rands/seeding/rand_unix.c

```diff
@@ -404,13 +408,16 @@ static ssize_t syscall_random(void *buf, size_t buflen)
     ERR_pop_to_mark();
     if (p_getentropy.p != NULL)
         return p_getentropy.f(buf, buflen) == 0 ? (ssize_t)buflen : -1;
-#  endif
+#    endif
+#  endif /* !__DragonFly__ */
 
     /* Linux supports this since version 3.17 */
 #  if defined(__linux) && defined(__NR_getrandom)
     return syscall(__NR_getrandom, buf, buflen, 0);
 #  elif (defined(__FreeBSD__) || defined(__NetBSD__)) && defined(KERN_ARND)
     return sysctl_random(buf, buflen);
+#  elif (defined(__DragonFly__)  && __DragonFly_version >= 500700)
```

Comment:
Is it worth conditioning this on the system call define as is done with `__NR_getrandom` above?

Reply:
As I understand the syscall for Linux is to avoid the mess with the presence or not of the getrandom function.
It is more straightforward here to avoid changing related linux check conditions, redefining __NR_getrandom while still have to check dragonflybsd version anyway.

Reply:
I wasn't suggesting a redefinition of __NR_getrandom, is there an equivalent define for DragonFlyBSD that could be used to detect the call more generally?

It was only a suggestion, so it if it doesn't fly, that's okay.

Reply:
Ah no sorry I did not have find any :-)